src/virtual_memory/anonymous.cpp

```diff
@@ -0,0 +1,114 @@
+/**
+ * @file
+ * @brief This file implements the virtual memory and virtual address handling
+ * @copyright Eta Scale AB. Licensed under the Eta Scale Open Source License. See the LICENSE file for details.
+ */
+
+#include <cerrno>
+#include <cstddef>
+#include <fcntl.h>
+#include <iostream>
+#include <string>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <sys/statvfs.h>
+#include <sys/types.h>
+#include <system_error>
+#include <unistd.h>
+#include "virtual_memory.hpp"
+
+namespace {
+	/* file constants */
+	/** @todo hardcoded start address */
+	const char* ARGO_START = (char*) 0x200000000000l;
+	/** @todo hardcoded end address */
+	const char* ARGO_END   = (char*) 0x600000000000l;
+	/** @todo hardcoded size */
+	const ptrdiff_t ARGO_SIZE = ARGO_END - ARGO_START;
+
+	/** @brief error message string */
+	const std::string msg_insufficient_memory = "ArgoDSM anonymous mappable memory is insufficient.";
+	/** @brief error message string */
+	const std::string msg_mprotect_fail = "ArgoDSM could not mprotect anonymous mapping";
+	/** @brief error message string */
+	const std::string msg_invalid_remap = "ArgoDSM encountered an invalid mapping attempt in virtual address space";
+	/** @brief error message string */
+	const std::string msg_main_mmap_fail = "ArgoDSM failed to set up virtual memory. Please report a bug.";
+
+	/* file variables */
+	/** @brief the address at which the backing storage space starts */
+	char* backing_addr;
+	/** @brief the current offset into the backing storage space */
+	std::size_t backing_offset;
+	/** @brief the offset within the memory region controlled to the beginning of the virtual address space */
+	std::size_t file_offset;
+}
+
+namespace argo {
+	namespace virtual_memory {
+		void init() {
+			/** @todo check desired range is free */
+			constexpr int flags = MAP_ANONYMOUS|MAP_SHARED|MAP_FIXED;
+			backing_addr = static_cast<char*>(
+				::mmap((void*)ARGO_START, ARGO_SIZE, PROT_NONE, flags, -1, 0)
```

Comment:
patch in 444383f, but we will need to check with cluster admins how to make this more robust.